autogpt/autogpt/agents/agent.py

```diff
@@ -180,7 +189,7 @@ async def propose_action(self) -> OneShotAgentActionProposal:
             ai_profile=self.state.ai_profile,
             ai_directives=directives,
             commands=function_specs_from_commands(self.commands),
-            include_os_info=self.legacy_config.execute_local_commands,
+            include_os_info=self.code_executor.config.execute_local_commands,
```

Comment:
Will it know the correct OS if I run Linux docker on windows or windows containers on windows

Reply:
It doesn't know such details, just the OS;
https://github.com/Significant-Gravitas/AutoGPT/blob/6ec708c771f786c7224b5aeb75e4bd9821f35985/autogpt/autogpt/agents/prompt_strategies/one_shot.py#L220

Reply:
@ntindle what is the issue with that? If it runs in a container the host OS does not matter, because we won't be running commands on the host OS from the container.

Reply:
Issue created